lib/purchases_flutter.dart

```diff
@@ -628,6 +636,30 @@ class Purchases {
   /// Android only. Call close when you are done with this instance of Purchases to disconnect
   /// from the billing services and clean up resources
   static Future<void> close() => _channel.invokeMethod('close');
+
+  static Future<Map<String, dynamic>>
+      _invokeMethodReturningPurchaserInfoJsonFromMap(String method,
+          // ignore: require_trailing_commas
+          [dynamic arguments]) async {
+    final response = await _invokeMethodReturningMap(
+      method,
+      arguments,
+    );
+    final purchaserInfoJson =
+        Map<String, dynamic>.from(response['purchaserInfo']);
+    return purchaserInfoJson;
+  }
+
+  static Future<Map<String, dynamic>> _invokeMethodReturningMap(String method,
+      // ignore: require_trailing_commas
+      [dynamic arguments]) async {
+    final result = await _channel.invokeMethod<Map<dynamic, dynamic>>(
+      method,
+      arguments,
+    );
+    final response = Map<String, dynamic>.from(result!);
```

Comment:
maybe we can log if result is nil?

Reply:
This will crash, and I think it should crash because something is very wrong if there's no result, so I don't think logging is necessary

Reply:
yeah i agree it should crash, just seems like we sometimes don't get the most descriptive error messages. i suppose using the ! should give us a clear message though